lib/moped/errors.rb

```diff
@@ -122,11 +122,12 @@ class PotentialReconfiguration < MongoError
       # not a master. (This encapsulates codes 10054, 10056, 10058)
       def reconfiguring_replica_set?
         err = details["err"] || details["errmsg"] || details["$err"] || ""
-        NOT_MASTER.include?(details["code"]) || err.include?("not master")
+        NOT_MASTER.include?(details["code"]) || err.include?("not master") || err.include?("could not contact primary")
       end
 
       def connection_failure?
-        CONNECTION_ERRORS_RECONFIGURATION.include?(details["code"])
+        err = details["err"] || details["errmsg"] || details["$err"] || ""
+        CONNECTION_ERRORS_RECONFIGURATION.include?(details["code"]) || err.include?("could not get last error") || err.include?("connection attempt failed") || err.include?("possible socket exception")
```

Comment:
Ok I removed the magic strings that had error codes in my examples and replaced them with error codes. However, there are two errors, the first of my examples and the second being #280, that did not get error codes back. @durran how are you handling that in the official driver? In this pull I check for strings which is brittle as @arthurnn mentions.